chemprop/data/splitting.py

```diff
@@ -251,3 +252,44 @@ def split_component(
             raise RuntimeError("Unreachable code reached!")
 
     return train, val, test
+
+
+def splits_from_file(
+    datapointss: Sequence[Union[Sequence[MoleculeDatapoint] | Sequence[ReactionDatapoint]]],
+    splits_file: Path,
```

Comment:
I use `Path` as the type hint, but it really could be any path-like class. A string path would be fine for example. In all the other cases we use `os.PathLike` as the type hint and I am unsure if a `Path` object passes this type check. I think either I should change mine to `PathLike` or we should find another way to type check path-like arguments.